File: furnitureshop-frontend/src/domain/http/httpEntity.ts
export class HttpEntity {
  private _authorization: string[];
  private _body: any;
  private _success: boolean;


  get authorization(): string[] {
    return this._authorization;
  }

  set authorization(value: string[]) {
    this._authorization = value;
  }

  get body(): any {
    return this._body;
  }

  set body(value: any) {
    this._body = value;
  }

  get success(): boolean {
    return this._success;
  }

  set success(value: boolean) {
    this._success = value;
  }
}

File: furnitureshop-backend/src/main/java/domain/http/HttpEntity.java
package domain.http;

import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

/**
 * A proxy object for request/response to UI.
 * @param <T>
 */
@Component
@Scope("prototype")
public class HttpEntity<T> {

    private String[] authorization;
    private T body;
    private boolean success = true;

    public HttpEntity() {
    }

    public String[] getAuthorization() {
        return authorization;
    }

    public void setAuthorization(String[] authorization) {
        this.authorization = authorization;
    }

    public HttpEntity(String[] authorization) {
        this.authorization = authorization;
    }

    public HttpEntity(T body) {
        this.body = body;
    }

    public HttpEntity(boolean success) {
        this.success = success;
    }

    public T getBody() {
        return body;
    }

    public void setBody(T body) {
        this.body = body;
    }

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }
}
